kamelet-support/src/main/java/io/kaoto/backend/metadata/parser/step/camelRoute/CamelRouteFileProcessor.java

```diff
@@ -0,0 +1,315 @@
+            final Map.Entry<String, Map<String, String>> parameter) {
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        StringParameter parsedParameter = new StringParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                parameterData.getOrDefault("defaultValue", null),
+                null
+        );
+
+        return parsedParameter;
+    }
+
+    private NumberParameter getNumberParameter(
+            final Map.Entry<String, Map<String, String>> parameter) {
+        //IntegerParameter doesn't fit some long types there are present
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        NumberParameter parsedParameter = new NumberParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                Long.parseLong(parameterData
+                        .getOrDefault("defaultValue", "0")
+                        .replaceAll("L", ""))
+        );
+
+        return parsedParameter;
+    }
+
+    private ObjectParameter getObjectParameter(
+            final Map.Entry<String, Map<String, String>> parameter) {
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        ObjectParameter parsedParameter = new ObjectParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                parameterData.getOrDefault("defaultValue", null)
+        );
+
+        return parsedParameter;
+    }
+
+    private BooleanParameter getBooleanParameter(
+            final Map.Entry<String, Map<String, String>> parameter) {
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        BooleanParameter parsedParameter = new BooleanParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                Boolean.parseBoolean(parameterData
+                        .getOrDefault("defaultValue", "false"))
+        );
+
+        return parsedParameter;
+    }
+
+    private String getStepType(final JsonObject component) {
+        final boolean canBeSource = component.getBoolean("producerOnly");
+        final boolean canBeSink = component.getBoolean("consumerOnly");
+
+        //If both are true or false we assume it to be an action.
+        //Or else, it has to be one of source or sink depending
+        //Take care that this later has to be duplicated because
+        //of the differences in how Camel components and kaoto components are
+        return canBeSource == canBeSink ? ACTION_TYPE
+                : canBeSource ? SOURCE_TYPE
+                : SINK_TYPE;
```

Comment:
Do you think this is harder to read than

```
if (canBeSource == canBeSink) {
  return ACTION_TYPE
} else if (canBeSource) {
  return SOURCE_TYPE
} else {
  return SINK_TYPE
}
```

I usually avoid ternary operators but I thought that for this case it is much more readable than a huge if else or a conditions map...

You can get on a single glance what is the default and what are the conditions for the others. Reminds me of a rust switch case like it is.

Reply:
yes it is. There is the need to search where are the separators : and ? using ternary operators. Especially given that the two : are indented at same level

Reply:
But I don't get why are you searching when the mapping are obvious, aka, which condition will produce which result...

I don't know, I was quite proud on how it was because I feel like it was really readable, but I'll just commit as you say if you disagree.